xmtp_id/src/associations/signature.rs

```diff
@@ -98,35 +124,51 @@ impl Signature for RecoverableEcdsaSignature {
     }
 }
 
-#[allow(dead_code)]
 #[derive(Debug, Clone)]
 pub struct Erc1271Signature {
     signature_text: String,
     signature_bytes: Vec<u8>,
     contract_address: String,
     block_number: u64,
+    chain_rpc_url: String,
 }
 
 impl Erc1271Signature {
     pub fn new(
         signature_text: String,
         signature_bytes: Vec<u8>,
         contract_address: String,
+        chain_rpc_url: String,
         block_number: u64,
     ) -> Self {
         Erc1271Signature {
             signature_text,
             signature_bytes,
             contract_address,
+            chain_rpc_url,
             block_number,
         }
     }
 }
 
+#[async_trait]
 impl Signature for Erc1271Signature {
-    fn recover_signer(&self) -> Result<MemberIdentifier, SignatureError> {
-        // TODO: Verify signature first
-        Ok(self.contract_address.clone().into())
+    async fn recover_signer(&self) -> Result<MemberIdentifier, SignatureError> {
+        let verifier = crate::erc1271_verifier::ERC1271Verifier::new(self.chain_rpc_url.clone());
+        // let runtime = Runtime::new().unwrap();
+        let is_valid = verifier
+            .is_valid_signature(
+                Address::from_slice(self.contract_address.as_bytes()), // TODO: `from_slice` will panic when input is not 20 bytes
```

Comment:
I'd expect the contract address to be a CAIP-10 address, at least according to the XIP.

Reply:
Here are my understanding of the flow:
- the platform SDK calls `getRequiredSignatures` and will need to pass in RPC URLs if required signatures contains an ERC1271 signature.
- We can do a sanity check that if the RPC URLs match the chain id specified in the CAIP-10.
- Use the RPC URL later on in the `recover_signer`.

I plan to break down the CAIP-10 string into pieces during deserialization .

WDYT?

Reply:
> the platform SDK calls getRequiredSignatures and will need to pass in RPC URLs if required signatures contains an ERC1271 signature.

I think the platform SDK is going to need to pass in a RPC URL at the time of client creation. Clients should be in control of their RPC provider, and the signer cannot be trusted to provide a valid `rpc_url` as part of signature generation (what if I created a malicious RPC provider that validated signatures which were not actually valid on the blockchain).

We can hack it for now by hardcoding a RPC URL as a constant in our code. 

> I plan to break down the CAIP-10 string into pieces during deserialization .

Sure, that sounds fine

Reply:
I think wallet apps will probably provide those RPC URLs after signing. Otherwise each inbox app will have to provide a set of node RPC URLs for all the chains where the wallets reside.

Reply:
PR #682 adds the CAIP-10 support